docs/guides/implementation-guides/how-to-implement-dependency-injection.md

```diff
@@ -3,13 +3,149 @@ id: how-to-implement-dependency-injection
 
-# How To Implement Dependency Injection
-
-Content in preparation.
+[Dependency injection (DI)](https://en.wikipedia.org/wiki/Dependency_injection) allows developers to write cleaner, more modular, and testable code. It accomplishes this by creating discrete services that are passed around/created as needed. It is especially useful when you are developing multi-platform applications that need specific service implementations for each platform.
 
 This guide will show you how to use Dependency Injection (DI) with _Avalonia UI_ and the MVVM pattern. 
 
-<img src={MvvmArchitectureDiagram} alt=""/>
+Let's assume that your MainViewModel has a dependency on IMyService. The constructor for ViewModel should looks like this:
+
+```csharp
+private readonly IMyService _myService;
+
+public MainViewModel(IMyService myService)
+{
+    _myService = myService;
+}
+```
+
+Typically you would directly instantiate `MyService` and pass it into `MainViewModel`, like this:
+
+```csharp
+var window = new MainWindow
+{
+    // Note: we assume that MyService implements IMyService
+    DataContext = new MainViewModel(new MyService())
+}
+```
+
+ This works great for simple constructors that are not used very often and don't change. But this technique does not scale very well because:
+- The more dependencies your constructor has the more things you will need to instantiate and pass in. Instantiating the dependencies locally (such as by doing new MainViewModel(new MyService())) results in direct rigid coupling to a specific instance of the dependencies.
+- Similarly if MainViewModel creates it's dependencies itself (such as in the constructor body) it also becomes directly coupled to the creation of the dependencies which can result in mostly the same problems. 
+- Furthermore, if the object is instantiated in many places, every single reference to any of the dependencies would also need to be updated should the dependencies of `MainViewModel` ever change (such as by requiring additional dependencies) or require a different implementation of a dependency. 
+
+Dependency injection solves these problem by abstracting away the creation of objects and their dependencies. This allows for well encapsulated services to be used and registered that will be automatically passed into any other service that is registered to use them. 
+
+See below the steps you need to do in order to resolve that dependencu using DI.
+
+## Step 1: Install the NuGet package for DI
+There are many dependency injection (DI) container providers available ([DryIoC](https://github.com/dadhi/DryIoc), [Autofac](https://github.com/autofac/Autofac), [Pure.DI](https://github.com/DevTeam/Pure.DI)) but this guide will only focus on `Microsoft.Extensions.DependencyInjection`. This is a lightweight, extensible dependency injection container that is part of the .NET Framework. It provides an easy-to-use and convention-based way to add DI to .NET applications, including Avalonia-based desktop applications.
```

Comment:
Slight correction. Microsoft DI is not part of the framework, which is why it's in an optional extension library. Also saying .NET Framework is misleading. .NET Framework specifically refers to the Windows only implementation of .NET, not the modern version of it which runs on more than just Windows.